### PR TITLE
What's new 2.11

### LINKED ISSUE #3401: 2.11 release notes
**Product:** Tarantool
**Since:** 2.11
**Audience/target:** all
**Root document:** new page in https://www.tarantool.io/en/doc/latest/release/
**SME:** @ Totktonada @ NeraverinTarantool 

# Details

Write release notes for Tarantool 2.11 in the form of What's New - complete human-readable descriptions of new features with examples and business value.
Use the [2.11.0-rc1 announcement](https://github.com/tarantool/tarantool/discussions/8371) as a starting point. 
Collaborate with the product team to form the final list of features to present in the release notes.

# Definition of done

- [ ] list of features to present in the release notes is approved by product managers
- [ ] the descriptions from the RC announcements are edited and possibly extended for correctness and clarity
- [ ] the release changelog on GitHub is linked from the Release notes

### PR DESCRIPTION
**Pull request by andreyaksenov**
Topic: [Releases > 2.11.x (LTS) > Tarantool 2.11.0](https://docs.d.tarantool.io/en/doc/2.11-release-notes/release/2.11.0/)

Notes about links:
- Links to Enterprise docs will start working after publishing Enterprise docs.
- The link to `Limit execution time` (in [this](https://docs.d.tarantool.io/en/doc/2.11-release-notes/release/2.11.0/#limitation-of-fiber-execution-time) section) will start working after merging the corresponding PR (https://github.com/tarantool/doc/pull/3448).

### CODE REVIEW

File: doc/release/2.11.0.rst [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+
+Released on 2023-05-22.
+
+*   Release: :tarantool-release:`2.11.0`.
+*   Tag: ``2.11.0``.
+
+
+The 2.11 release of Tarantool includes many new features and fixes for Enterprise and Community versions. 2.11 is the long-term support (LTS) release with four years of maintenance. This means that you will continue to receive all the necessary security fixes and bug fixes throughout this period. You can learn more about the Tarantool release policy from the :doc:`corresponding document </release/policy>`.
+
+
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: I would extend the introduction a bit: 
+ "find the detailed changelog on GitHub #link#"
+ "Read upgrade instructions in docs #link#"

And also add a ToC to make feature names more visible from the start.

File: doc/release/2.11.0.rst [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+        params = {
+            transport = 'ssl',
+            ssl_key_file = '/path_to_key_file',
+            ssl_cert_file = '/path_to_cert_file',
+            ssl_ciphers = 'HIGH:!aNULL',
+            ssl_password = 'topsecret'
+        }
+    }}
+
+
```
Comment (p7nov) [inline doc/release/2.11.0.rst:80]: Maybe add a "Learn more" sentence for consistency? There are ones in other subsections.

File: doc/release/2.11.0.rst [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+Security enhancements
+~~~~~~~~~~~~~~~~~~~~~
+
+
+.. _encrypted_ssl_tls_keys:
+
+Encrypted SSL/TLS keys
+**********************
+
+Tarantool Enterprise now supports `encrypted <https://www.tarantool.io/en/enterprise_doc/security/#traffic-encryption>`_ SSL/TLS private key files protected with a password.
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: It's better to create an intersphinx mapping to avoid absolute links to the Enterprise doc. Although this may be out of this PR's scope.

File: doc/release/2.11.0.rst [hunk 4]
```diff
@@ -0,0 +1,586 @@
...
+        }
+    }}
+
+
+.. _security_enforcement_options:
+
+Security enforcement options
+****************************
+
+With v2.11, Tarantool Enterprise includes new security enforcement options.
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: "v2.11"  here and below seems like tech jargon to me, I would avoid this spelling.
Just "2.11" is fine for such cases.

File: doc/release/2.11.0.rst [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+These options provide the ability to improve security by flexible customization of a password policy.
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: This is hard to read – too many connections.

File: doc/release/2.11.0.rst [hunk 6]
```diff
@@ -0,0 +1,586 @@
...
+A password policy enables you to enforce the use of strong passwords, set up a maximum password age, and so on.
+For example, the ``password_min_length`` configuration option specifies the minimum number of characters for a password:
```
Comment (p7nov) [inline doc/release/2.11.0.rst:88]: This whole description seems a bit too wordy and detailed for a What's New page.
To me, this would be enough:
```
A password policy enables you to enforce the use of strong passwords, set up a maximum password age, and so on.
A couple examples:
* the minimum password length:
  #code#
*  the maximum password lifetime: 
  #code#   
Note that by default, new options are not specified.
You can learn more about all the available options from ...
```

File: doc/release/2.11.0.rst [hunk 7]
```diff
@@ -0,0 +1,586 @@
...
+.. _pap_sha_256:
+
+PAP-SHA256 authentication method
+********************************
+
+By default, Tarantool uses the CHAP protocol to authenticate users and applies SHA-1 hashing to passwords.
+In this case, password hashes are stored in the ``_user`` space unsalted.
+If an attacker gains access to the database, they may crack a password, for example, using a rainbow table.
+
+With Tarantool Enterprise, you can enable PAP authentication with the SHA256 hashing algorithm.
```
Comment (p7nov) [inline doc/release/2.11.0.rst:115]: Same as above: can we express the essence with less words? I don't think that we need a second mentioning of cracking and rainbow tables.

File: doc/release/2.11.0.rst [hunk 8]
```diff
@@ -0,0 +1,586 @@
...
+
+
+.. _read_views:
+
+Read views
+~~~~~~~~~~
+
+Starting with v2.11, Tarantool Enterprise provides the ability to create read views - in-memory snapshots of the entire database that aren't affected by future data modifications.
+Read views can be used to make complex analytical queries.
+This reduces the load on the main database and improves RPS for a single Tarantool instance.
```
Comment (p7nov) [inline doc/release/2.11.0.rst:138]: I doubt about RPS - isn't it too technical or jargon? Maybe just performance?

File: doc/release/2.11.0.rst [hunk 9]
```diff
@@ -0,0 +1,586 @@
...
+For example, you can enable storing old and new tuples for each write-ahead log record.
+This is especially useful for the ``update`` operation because a write-ahead log record contains only a key value.
+
+See the `WAL extensions <https://www.tarantool.io/en/enterprise_doc/wal_extensions/>`_ topic to learn how to enable and configure WAL extensions.
+
+
+
+.. _community:
+
+Community
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: maybe use the full product name?
Just "Community" as a heading is ambiguous. Is it about people who use Tarantool?

File: doc/release/2.11.0.rst [hunk 10]
```diff
@@ -0,0 +1,586 @@
...
+The oldest release available for downgrade is :doc:`2.8.2 </release/2.8.2>`.
+
+
+
+.. _new_bootstrap_strategy:
+
+New bootstrap strategy
+~~~~~~~~~~~~~~~~~~~~~~
+
+In previous Tarantool versions, the :ref:`replication_connect_quorum <cfg_replication-replication_connect_quorum>` option was used to specify the number of nodes to be up and running to start a replica set.
```
Comment (p7nov) [inline doc/release/2.11.0.rst]:
```suggestion
In previous Tarantool versions, the :ref:`replication_connect_quorum <cfg_replication-replication_connect_quorum>` option was used to specify the number of running nodes to start a replica set.
```
Just a shortening idea)

File: doc/release/2.11.0.rst [hunk 11]
```diff
@@ -0,0 +1,586 @@
...
+This option was designed to simplify a replica set bootstrap.
+But in fact, this behaviour brought some issues during a cluster lifetime and maintenance operations, for example:
```
Comment (p7nov) [inline doc/release/2.11.0.rst]:
```suggestion
But in fact, this behavior brought some issues during a cluster lifetime and maintenance operations, for example:
```
Let's stick to the US spelling

File: doc/release/2.11.0.rst [hunk 12]
```diff
@@ -0,0 +1,586 @@
...
+
+*   Users who didn't change this option encountered problems with the partial cluster bootstrap.
+*   Users who changed the option encountered problems during the instance restart.
+
+With v2.11, ``replication_connect_quorum`` is deprecated in favor of :ref:`bootstrap_strategy <cfg_replication-bootstrap_strategy>`.
+This option works during a replica set bootstrap and implies sane default values for other parameters based on the replica set configuration.
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: "sane" is generally not welcome by big tech [style guides](https://developers.google.com/style/word-list#letter-s).

File: doc/release/2.11.0.rst [hunk 13]
```diff
@@ -0,0 +1,586 @@
...
+This might be useful in the case of a failed upgrade.
+
+To prepare a database for using it on an older Tarantool instance, call ``box.schema.downgrade`` and pass the desired Tarantool version:
+
+.. code-block:: tarantoolsession
+
+    tarantool> box.schema.downgrade('2.8.4')
+
+To see Tarantool versions available for downgrade, call ``box.schema.downgrade_versions()``.
+The oldest release available for downgrade is :doc:`2.8.2 </release/2.8.2>`.
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: "old/older" referring to versions is not recommended by common style guides (see [Google](https://developers.google.com/style/word-list#letter-o)).

File: doc/release/_images/wal_extensions.svg
Comment (p7nov) [inline doc/release/_images/wal_extensions.svg:1]: We have a [kind-of-UI-kit](https://www.tarantool.io/en/doc/latest/contributing/docs/images/) (it's pretty limited) for building diagrams that fit the docs style, please use it.

File: doc/release/2.11.0.rst [hunk 14]
```diff
@@ -0,0 +1,586 @@
...
+
+
+To enable table scanning explicitly, use the new ``SEQSCAN`` keyword:
+
+.. code-block:: sql
+
+    SET SESSION "sql_seq_scan" = false;
+    SELECT a FROM SEQSCAN t WHERE a + 1 > 10;
+
+Starting with v3.0, ``SEQSCAN`` will be required for scanning queries with the ability to disable the check using the ``sql_seq_scan`` session setting.
```
Comment (p7nov) [inline doc/release/2.11.0.rst]: Pls avoid future promises with exact dates or versions.

File: doc/release/calendar.rst [hunk 15]
```diff
@@ -30,6 +30,10 @@ End of series support means the release series will no longer receive any patche
             -   Release date
             -   End of series support
 
+        *   -   :doc:`2.11.0 LTS </release/2.11.0>`
+            -   May 22, 2023
+            -   Not planned yet
+
         *   -   :doc:`2.10.6 </release/2.10.6>`
```
Comment (p7nov) [inline doc/release/calendar.rst:37]: We should also check and update the end of support for currently supported series. May be out of scope here.

**Review by p7nov — CHANGES_REQUESTED** [review]
Generally, I would:
- add more common release information to the intro
- try to shorten feature descriptions that almost completely repeat the corresponding docs.

Find details in comments.

File: doc/release/2.11.0.rst (hunk 4 above)
Comment (p7nov) [reply doc/release/2.11.0.rst]: Also, shouldn't it be complete "2.11.0"?

Comment (andreyaksenov) [reply doc/release/2.11.0.rst]: I think it's okay to use the shorter version here.

File: doc/release/2.11.0.rst (hunk 5 above)
Comment (andreyaksenov) [reply doc/release/2.11.0.rst]: combined the first and second sentences to make it simpler

File: doc/release/2.11.0.rst (hunk 8 above)
Comment (andreyaksenov) [reply doc/release/2.11.0.rst:138]: I think it's okay here because `RPS` is more specific than `performance`.

File: doc/release/2.11.0.rst (hunk 9 above)
Comment (andreyaksenov) [reply doc/release/2.11.0.rst]: Agree, fixed both Enterprise and Community titles

File: doc/release/2.11.0.rst (hunk 12 above)
Comment (andreyaksenov) [reply doc/release/2.11.0.rst]: thanks, changed to `sensible`

File: doc/release/2.11.0.rst (hunk 13 above)
Comment (andreyaksenov) [reply doc/release/2.11.0.rst]: changed to `earliest`

File: doc/release/_images/wal_extensions.svg
Comment (andreyaksenov) [reply doc/release/_images/wal_extensions.svg:1]: thanks, updated

File: doc/release/calendar.rst (hunk 15 above)
Comment (andreyaksenov) [reply doc/release/calendar.rst:37]: > * try to shorten feature descriptions that almost completely repeat the corresponding docs.

Yep, I totally agree with this, but I tried to find some balance regarding the provided details for documented and undocumented features. For example, for features that already have docs, a what's new description might be too short compared to undocumented features. And a what's new entry cannot be too short for an undocumented feature because it's the only information about this feature in docs.

File: doc/release/2.11.0.rst (hunk 14 above)
Comment (andreyaksenov) [reply doc/release/2.11.0.rst]: btw, compat docs already give this promise: https://www.tarantool.io/en/doc/latest/reference/reference_lua/compat/sql_seq_scan_default/
Not sure what is the best way to rephrase this..

File: doc/release/calendar.rst (hunk 15 above)
Comment (andreyaksenov) [reply doc/release/calendar.rst:37]: > We should also check and update the end of support for currently supported series. May be out of scope here.

as I understand, it should be a four-year period starting from the Release date

Comment (andreyaksenov) [reply doc/release/calendar.rst:37]: > And also add a ToC to make feature names more visible from the start.

I'm not a hundred percent sure this is required - this list occupies a lot of space and mostly duplicates a topic's TOC on the right.

File: doc/release/2.11.0.rst (hunk 14 above)
Comment (Totktonada) [reply doc/release/2.11.0.rst]: Sure, this is compat's mechanics: if a major version X introduces an option as 'old', then the major version X+1 changes it to 'new'. In context of the major version 3 it is on review: https://github.com/tarantool/tarantool/pull/8602.

File: doc/release/2.11.0.rst (hunk 1 above)
Comment (p7nov) [reply doc/release/2.11.0.rst]: I still feel that intro is a bit too brief. The _What's New_ format is new for Tarantool, and so is the combination of CE and EE features in one release document. Thus, to avoid confusion, I would write **explicitly** that:
- this doc provides an overview of important features
- here we describe both CE and EE features
- the complete list of changes and bug fixes now lives only on Github

Also, more words about the upgrade might be useful. A note looks like it's some slightly related info while it's pretty important. I would place the upgrade info either under a subheading or just in a separate paragraph of the intro. Plus, we can think about being more helpful and provide some guidance on upgrade issues, smth like: "file us an issue" or "reach out to support" - PM team should know better.

**Review by p7nov — APPROVED** [review]
I still have some thoughts about the intro, please see the new comment.
Otherwise LGTM.

File: doc/release/2.11.0.rst (hunk 14 above)
Comment (andreyaksenov) [reply doc/release/2.11.0.rst]: @Totktonada thanks for clarification!

**Comment by andreyaksenov** [comment]
> this doc provides an overview of important features

agree, totally forgot to add what this doc is about :(

> here we describe both CE and EE features

The second sentence will be: `This document provides an overview of the most important features for Enterprise and Community versions.`

> the complete list of changes and bug fixes now lives only on Github

The document already contains a link to GitHub. Not sure we should mention `only` right in the docs. I think such info is good for announcements, blots, etc. but not for the doc itself.

> Also, more words about the upgrade might be useful. A note looks like it's some slightly related info while it's pretty important.

Yep, it's very important, but I'd not mix this kind of 'marketing' overview topic with the detailed upgrade notes. 

> Plus, we can think about being more helpful and provide some guidance on upgrade issues, smth like: "file us an issue" or "reach out to support" - PM team should know better.

Maybe we can create a separate `Support` page in docs with all support channels? So we can mention it along with the `Upgrades` topic in `What's new` documents. I still feel that while this information is important, users open the What's new page to learn about new features in the first place.
